pkg/deps/types/accquerier.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"context"
+
+	"cosmossdk.io/depinject"
+	"github.com/cosmos/cosmos-sdk/client"
+	accounttypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	"github.com/pokt-network/poktroll/pkg/relayer"
+)
+
+type AccountQuerier interface {
+	GetAccount(ctx context.Context, address string) (accounttypes.AccountI, error)
+}
+
+type accQuerier struct {
```

Comment:
Following-on from the comment above, I would suggest moving this to a `pkg/client/query` package.

In general, we should (strongly) prefer putting interfaces in distinct packages from their implementations so as to avoid any potential for import cycles between them.

E.g. because the `Notification` interface is defined next to its implementation, `user` pkg has to import it, creating a cycle:
```go
// user/user.go
package user

import (
    "example/notification"
)

// User struct
type User struct {
    Name string
}

// SendNotification function that takes a Notification interface
func SendNotification(u User, n notification.Notification) {
    n.Notify(u)
}
```

```go
// notification/notification.go
package notification

import (
    "example/user"
    "fmt"
)

// Notification interface
type Notification interface {
    Notify(u user.User)
}

// UserNotification struct implementing Notification
type UserNotification struct{}

// Notify method for UserNotification
func (n UserNotification) Notify(u user.User) {
    fmt.Printf("Notifying user %s\n", u.Name)
}
```